src/suit.c

```diff
@@ -316,14 +316,32 @@ int suit_processor_get_manifest_metadata(uint8_t *envelope_str, size_t envelope_
 		ret = cbor_decode_SUIT_Digest(digest_bstr->value, digest_bstr->len, &digest_cbor,
 					      &bytes_processed);
 
-		/* The SHA256 algorithm is enforced by CDDL, so the digest length is known. */
-		if ((bytes_processed != digest_bstr->len) ||
-		    (digest_cbor._SUIT_Digest_suit_digest_bytes.len != 32)) {
+		if (bytes_processed != digest_bstr->len) {
 			ret = SUIT_ERR_DECODING;
```

Comment:
Considering in all cases returned error is the same, why only store the error code and not return?

Reply:
To have a common clean-up code:
```
decoder_state->step = INVALID;
```

Reply:
Then shouldn't all the following if clauses do the same?

Reply:
Ah, now I understand - all other branches aligned! :+1: